### PR TITLE
[To Vector] Fix Tablet.addValue()

### PR DESCRIPTION
**Pull request by samperson1997**
Fix `Tablet.addValue()` and add another method to insertTablet

### CODE REVIEW

File: tsfile/src/main/java/org/apache/iotdb/tsfile/write/record/Tablet.java [hunk 1]
```diff
@@ -121,13 +121,11 @@ public void addValue(String measurementId, int rowIndex, Object value) {
     IMeasurementSchema measurementSchema = schemas.get(indexOfValue);
 
     if (measurementSchema.getType().equals(TSDataType.VECTOR)) {
-      for (int i = 0; i < measurementSchema.getValueMeasurementIdList().size(); i++) {
-        TSDataType dataType = measurementSchema.getValueTSDataTypeList().get(i);
-        addValueOfDataType(dataType, rowIndex, measurementIndex.get(measurementId), value);
-      }
+      int indexInVector = measurementSchema.getMeasurementIdColumnIndex(measurementId);
+      TSDataType dataType = measurementSchema.getValueTSDataTypeList().get(indexInVector);
+      addValueOfDataType(dataType, rowIndex, indexOfValue + indexInVector, value);
```
Comment (HTHou) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/write/record/Tablet.java:126]: We need to consider when `value == null`, init the `BitMaps` in tablet if needed, set an irrelevant value in the array, and mark this position of bitMap.

Comment (HTHou) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/write/record/Tablet.java:126]: Fixed by myself...

File: session/src/main/java/org/apache/iotdb/session/Session.java [hunk 2]
```diff
@@ -1541,7 +1541,7 @@ protected void sortTablet(Tablet tablet) {
         int measurementSize = schema.getValueMeasurementIdList().size();
         for (int j = 0; j < measurementSize; j++) {
           tablet.values[columnIndex] =
-              sortList(tablet.values[columnIndex], schema.getValueTSDataTypeList().get(i), index);
+              sortList(tablet.values[columnIndex], schema.getValueTSDataTypeList().get(j), index);
```
[the remark this answers is not in the document]
Comment (samperson1997) [reply session/src/main/java/org/apache/iotdb/session/Session.java:1544]: Added. Also add a complicated test case for mix MeasurementSchema and VectorMeasurementSchema in the same tablet : )

File: example/session/src/main/java/org/apache/iotdb/VectorSessionExample.java [hunk 3]
```diff
@@ -215,18 +215,42 @@ private static void insertTabletWithAlignedTimeseries()
...
+      }
+      timestamp++;
+    }
+
+    if (tablet.rowSize != 0) {
+      session.insertTablet(tablet);
+      tablet.reset();
+    }
+
+    // Method 2 to add tablet data
```
[the remark this answers is not in the document]
Comment (samperson1997) [reply example/session/src/main/java/org/apache/iotdb/VectorSessionExample.java]: Splitted

**Review by HTHou — APPROVED** [review]

**Pull request merged**